### PR TITLE
Dev-5285 Covid-19 Page Scaffold

### PR DESCRIPTION
**Pull request by jonhill13**
**High level description:**

Implements the Covid-19 Page Outline.

**Technical details:**

> • based off of the agency page ( Great Work @MaxwellKendall )

> • new route based off of Covid-19 release

> • new container

> • new helpers

> • new data mapping

**JIRA Ticket:**
[DEV-5285](https://federal-spending-transparency.atlassian.net/browse/DEV-5285)

**Mockup:**
https://invis.io/RYA3XN5WP#/273832670_Homepage_2-2_E

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [x] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
- [x] Verified cross-browser compatibility
- [x] Verified mobile/tablet/desktop/monitor responsiveness
- [N/A] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [N/A] Added Unit Tests for methods in Container Components, reducers, and helper functions `if applicable`
- [x] All componentWillReceiveProps, componentWillMount, and componentWillUpdate in relevant child/parent components/containers replaced with [future compatible life-cycle methods](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html) per [JIRA item 3339](https://federal-spending-transparency.atlassian.net/browse/DEV-3339)
- [N/A] [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
- [N/A] [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
- [N/A] Design review complete `if applicable`
- [N/A] [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [ ] Code review complete

### CODE REVIEW

File: src/_scss/components/_footerLinkToAdvancedSearch.scss [hunk 1]
```diff
@@ -26,7 +26,7 @@
                 font-weight: $font-semibold;
             }
         }
-        .agency-search-button {
+        .footerLinkToAdvancedSearch-search-button {
```
Comment (jameslee40) [inline src/_scss/components/_footerLinkToAdvancedSearch.scss:29]: Not too familiar with [BEM](http://getbem.com/naming/), but shouldn't this be `.footerLinkToAdvancedSearch__search-button`? Or, just `.footerLinkToAdvancedSearch__button`? I guess I'm confused as to when we should be using `-` versus `__` and when we should be using `camel case` for scss.

File: src/_scss/pages/covid19/index.scss [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+      .sidebar {
+          display: none;
+          @media(min-width: $medium-screen) {
+              @include display(flex);
+              @include flex-direction(column);
+              width: 22%;
+              margin: rem(20);
+          }
+          .covid19-sidebar-reference {
+              display: none;
```
Comment (jameslee40) [inline src/_scss/pages/covid19/index.scss:132]: `@include display(none)`

File: src/_scss/pages/covid19/index.scss [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+              &.float-sidebar {
+                  display: block;
```
Comment (jameslee40) [inline src/_scss/pages/covid19/index.scss:134]: `@include display(block)`

File: src/_scss/pages/covid19/index.scss [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+          }
+      }
+      .body {
+          @import "components/_footerLinkToAdvancedSearch";
+          @include display(flex);
+          width: 100%;
+          @include flex-direction(column);
+          padding: 2.5% 2.5% 0 2.5%;
+          margin: rem(20) 0 rem(20) 0;
+          background: white;
```
Comment (jameslee40) [inline src/_scss/pages/covid19/index.scss:193]: `$color-white`

File: src/_scss/pages/covid19/index.scss [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+              @import '../agency/v2/tabs';
+              @include display(flex);
+              @include flex-wrap(wrap);
+              margin-bottom: $global-mrg;
+              .body__header {
+                  width: 100%;
+                  display: flex;
+                  flex-wrap: wrap;
+                  align-items: center;
+                  justify-content: space-between;
```
Comment (jameslee40) [inline src/_scss/pages/covid19/index.scss:281]: `@include display(flex)`
`@include flex-wrap(wrap)`
etc...

File: src/js/components/sharedComponents/FooterLinkToAdvancedSearch.jsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    description: PropTypes.string
+};
+
+const FooterLinkToAdvancedSearch = ({
+    clickedSearch,
+    title,
+    description
+}) => (
+    <div className="footerLinkToAdvancedSearch">
+        <div className="footerLinkToAdvancedSearch-content">
```
Comment (jameslee40) [inline src/js/components/sharedComponents/FooterLinkToAdvancedSearch.jsx:21]: Comment above about [BEM](http://getbem.com/naming/), may need to update this to `footerLinkToAdvancedSearch__content`.

File: src/js/components/sharedComponents/FooterLinkToAdvancedSearch.jsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+            <h4>{title}</h4>
+            <p>
+                Check out the <strong>Award Search</strong> page <br />
+                {description}
+            </p>
+            <button
+                className="footerLinkToAdvancedSearch-search-button"
```
Comment (jameslee40) [inline src/js/components/sharedComponents/FooterLinkToAdvancedSearch.jsx:28]: Comment above about [BEM](http://getbem.com/naming/), may need to update this to `footerLinkToAdvancedSearch__button` or `footerLinkToAdvancedSearch__search-button`.

File: src/js/containers/covid19/Covid19Container.jsx [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+    // const DEFOptions = getDEFOptions(setselectedDEF, defaultSortFy);
+
+    const jumpToCovid19Section = (section) => jumpToSection(section, activeSection, setActiveSection);
+
+    return (
+        <div className="usa-da-covid19-page">
+            <MetaTags {...covidPageMetaTags} />
+            <Header />
+            <StickyHeader>
+                <>
```
Comment (jameslee40) [inline src/js/containers/covid19/Covid19Container.jsx:46]: For some reason this `<` is showing up red for me?

File: src/_scss/pages/covid19/index.scss (hunk 5 above)
Comment (jonhill13) [reply src/_scss/pages/covid19/index.scss:281]: I thought we were moving away from @include because it was part of bourbon or something?

Comment (jameslee40) [reply src/_scss/pages/covid19/index.scss:281]: I was not aware. Please disregard my comments then!

File: src/js/containers/covid19/Covid19Container.jsx (hunk 8 above)
Comment (jonhill13) [reply src/js/containers/covid19/Covid19Container.jsx:46]: Not sure, it is not for me.

Comment (jameslee40) [reply src/js/containers/covid19/Covid19Container.jsx:46]: Yeah, the page renders fine, so I think we're good!

File: src/_scss/components/_footerLinkToAdvancedSearch.scss (hunk 1 above)
Comment (jonhill13) [reply src/_scss/components/_footerLinkToAdvancedSearch.scss:29]: made it this footerLinkToAdvancedSearch__button

File: src/_scss/pages/covid19/index.scss (hunk 2 above)
Comment (jonhill13) [reply src/_scss/pages/covid19/index.scss:132]: same as my other comment below

File: src/_scss/pages/covid19/index.scss (hunk 3 above)
Comment (jonhill13) [reply src/_scss/pages/covid19/index.scss:134]: same as my other comment below

File: src/_scss/pages/covid19/index.scss (hunk 5 above)
Comment (jonhill13) [reply src/_scss/pages/covid19/index.scss:281]: same as my other comment below

**Pull request merged**